src/global/components/Navbar.tsx

```diff
@@ -0,0 +1,91 @@
+import { useLayoutEffect } from "react";
+import { NavLink } from "react-router-dom";
+import { useRecoilState, useRecoilValue } from "recoil";
+import { menuState, mobileState } from "../state/navbarState";
+import MenuDropdown from "./MenuDropdown";
+import { pathSelector } from "../state/navigationState";
+
+export default function Navbar() {
+  const [isMenuOpen, setMenuOpen] = useRecoilState(menuState);
+  const filteredPathsArray = useRecoilValue(pathSelector);
+  const [isMobileSize, setIsMobileSize] = useRecoilState(mobileState);
+
+  const handleOpenMenu = (e: React.MouseEvent<HTMLButtonElement>) => {
+    setMenuOpen(!isMenuOpen);
+  };
+
+  useLayoutEffect(() => {
+    if (window.innerWidth < 768) setIsMobileSize(true);
+  });
+
+  window.addEventListener("resize", () => {
+    if (window.innerWidth < 768 && !isMobileSize) setIsMobileSize(true);
+    if (window.innerWidth >= 768 && isMobileSize) setIsMobileSize(false);
+  });
```

Comment:
You shouldn't write event listeners outside `useEffect` it will cause to much rerendering. Write it in the useEffect and then also add a cleanup function

Reply:
Removed completely.